sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4773,10 +4777,23 @@ enqueueMemFillHelper(pi_command_type CommandType, pi_queue Queue, void *Ptr,
   // Get a new command list to be used on this call
   ze_command_list_handle_t ZeCommandList = nullptr;
   ze_fence_handle_t ZeFence = nullptr;
+
   // Performance analysis on a simple SYCL data "fill" test shows copy engine
   // is faster than compute engine for such operations.
+  bool PreferCopyEngine = Queue->Device->hasCopyEngine();
+  if (PreferCopyEngine) {
```

Comment:
Two questions:
1. Why are we checking only when we want to use Copy Engine? Should we not be checking this irrespective of whether we want to use copy or compute engine?
2. Also, Are we checking if the pattern size is a power of two anywhere?

Reply:
1. Here I am mostly checking if to use copy engine or not. I can rework the code that the assertion that the chosen queue can support the size is performed always.
2. Don't know, will add the check.

Reply:
we actually had the pow-2 check already later in the code, moved it